vshard/replicaset.lua

```diff
@@ -75,6 +75,108 @@ local fiber_cond_wait = util.fiber_cond_wait
 local future_wait = util.future_wait
 local gsc = util.generate_self_checker
 
+--
+-- vconnect is asynchronous vshard greeting, saved inside netbox connection.
+-- It stores future object and additional info, needed for its work.
+-- Future is initialized, when the connection is established (inside
+-- netbox_on_connect). The connection cannot be considered "connected"
+-- until vconnect is properly validated.
+--
+local function conn_set_vconnect(conn)
+    assert(conn.replica ~= nil)
+    conn.vconnect = {
+        -- Whether the connection is done to the named replica.
+        is_named = conn.replica.id == conn.replica.name,
+        -- Used to wait for the appearance of vconnect.future object,
+        -- if call is done before the connection is established.
+        future_cond = fiber.cond(),
+    }
+    local vconn = conn.vconnect
+    if not vconn.is_named then
+        -- Skip validation without any waiting.
+        conn.vconnect.is_validated = true
```

Comment:
I suggest to simply do `conn.vconnect = nil`.

Reply:
Done, but it requires calling `conn_vconnect_set` inside `on_connect` trigger.